### PR TITLE
[feat] Add gov handler to create CL <> Gamm Pool link

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #XXX

## What is the purpose of the change
Add gov handler to create CL <> Gamm Pool link

## Testing and Verifying
added test

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/gamm/keeper/migrate.go [hunk 1]
```diff
@@ -354,3 +354,33 @@ func (k Keeper) UpdateMigrationRecords(ctx sdk.Context, records []types.Balancer
 	}
 	return nil
 }
+
+func (k Keeper) LinkBalancerPoolWithClPoolRecords(ctx sdk.Context, records []types.BalancerToConcentratedPoolLink) error {
+	// TODO: check case where balancePoolExist but ClPool doesnot exist
```
Comment (stackman27) [inline x/gamm/keeper/migrate.go:359]: wanted to get thought on this 

- i think the frontend team just wants to link a CL pool to GAMM pool considering same denom. would this be necessary then?

**Comment by jonator** [comment]
Wait I thought @czarcas7ic already implemented this? As I was told? Unless I'm missing something

**Comment by mattverse** [comment]
@jonator no, you are correct, I think there was some sort of miscommunication that happened for this PR?

**Comment by mattverse** [comment]
@stackman27 How is this different from `ReplaceMigrationRecordsProposal`? I think we can basically do the same thing with `ReplaceMigrationRecordsProposal`?

**Comment by czarcas7ic** [comment]
Replace and UpdateMigrationRecordsProposal should be doing the same thing as this, yes

**Comment by stackman27** [comment]
@mattverse @czarcas7ic i dont think replace and update creates a new CL pool for a given balancer pool and links them. 

this PR updates the pool link if both pools exists and it also creates a new link if one exist and other dont. I thought that was the source of @jonator issue

**Comment by czarcas7ic** [comment]
I don't think there was any code that needed to be written, I think Jon just needed someone to submit a prop to link two existing pools

**Comment by stackman27** [comment]
> I don't think there was any code that needed to be written, I think Jon just needed someone to submit a prop to link two existing pools

i see, i didn't realize there was a CL pool that existed for the same denom already hence, just triggering gov prop helped. But what about cases where there is a CL pool with the same denom doesnot exist for a given balancer pool. Would this be helpful there?

**Comment by mattverse** [comment]
> i see, i didn't realize there was a CL pool that existed for the same denom already hence, just triggering gov prop helped. But what about cases where there ia CL pool with the same denom doesnot exist for a given balancer pool. Would this be helpful there?

wdym? If you are referring to a case where Balancer pool denom and CL pool denom does not match, would that case not lead to migration?

**Comment by stackman27** [comment]
> > i see, i didn't realize there was a CL pool that existed for the same denom already hence, just triggering gov prop helped. But what about cases where there ia CL pool with the same denom doesnot exist for a given balancer pool. Would this be helpful there?
> 
> wdym? If you are referring to a case where Balancer pool denom and CL pool denom does not match, would that case not lead to migration?

i see, talked to john and dont think this is necessary anymore. Closing it

**Pull request closed without merging**